etree.go

```diff
@@ -1236,9 +1240,12 @@ func (e *Element) CreateAttr(key, value string) *Attr {
 }
 
 // createAttr is a helper function that creates attributes.
-func (e *Element) createAttr(space, key, value string, parent *Element) *Attr {
+func (e *Element) createAttr(space, key, value string, parent *Element, settings ...ReadSettings) *Attr {
```

Comment:
I'm not sure why you're passing settings as a vararg. Also I don't think it's necessary to pass settings at all.

I think this would be a better approach:
```go
func (e *Element) createAttr(space, key, value string, parent *Element, preserveDups bool) *Attr {
	if !preserveDups {
		for i, a := range e.Attr {
			if space == a.Space && key == a.Key {
				e.Attr[i].Value = value
				return &e.Attr[i]
			}
		}
	}
    ...
```